test/e2e/util/mongotester/mongotester.go

```diff
@@ -135,7 +136,33 @@ func (m *Tester) HasTlsMode(tlsMode string, tries int, opts ...OptionApplier) fu
 	return m.hasAdminParameter("sslMode", tlsMode, tries, opts...)
 }
 
-func (m *Tester) hasAdminParameter(key string, expectedValue interface{}, tries int, opts ...OptionApplier) func(t *testing.T) {
+// CustomRolesResult is a type to decode the result of getting rolesInfo.
+type CustomRolesResult struct {
+	Roles []automationconfig.CustomRole
+}
+
+func (m *Tester) VerifyRoles(expectedRoles []automationconfig.CustomRole, tries int, opts ...OptionApplier) func(t *testing.T) {
+	return m.hasAdminCommandResult(func(m *Tester, t *testing.T) bool {
```

Comment:
[q] what is the reason for the `m` in the function signature `func(m *Tester, t *testing.T)` ?

It looks like the functions that use this are already on the `m *Tester` type and already have access to `m`.

Reply:
You're right, its superfluous. I have removed it. Good catch!